pkg/private/install.py.tpl

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This template is completed by `pkg_install` to create installation scripts,
+# and will not function on its own.  See pkg/install.bzl for more details.
+
+import argparse
+import json
+import logging
+import os
+import shutil
+import sys
+
+import private.manifest as manifest
+
+# Globals used for runfile path manipulation.
+#
+# These are necessary because runfiles are different when used as a part of
+# `bazel build` and `bazel run`. # See also
+# https://docs.bazel.build/versions/4.1.0/skylark/rules.html#tools-with-runfiles
+
+# Bazel's documentation claims these are set when calling `bazel run`, but not other
+# modes, like in `build` or `test`.  We'll see.
+CALLED_FROM_BAZEL_RUN = bool(os.getenv("BUILD_WORKSPACE_DIRECTORY") and
+                             os.getenv("BUILD_WORKING_DIRECTORY"))
+
+WORKSPACE_NAME = "##WORKSPACE_NAME##"
+# This seems to be set when running in `bazel build` or `bazel test`.
+RUNFILE_PREFIX = os.path.join(os.getenv("RUNFILES_DIR"), WORKSPACE_NAME) if os.getenv("RUNFILES_DIR") else None
```

Comment:
Is it an error if RUNFILE_PREFIX is None and CALLED_FROM_BAZEL_RUN is false?

Reply:
Yes.  However, I would not be surprised if this variable isn't set when in Windows, and was causing the test failures there.  I added a TODO mentioning #387.

Reply:
Is now an error as of 69c993b